packages/runtime/src/components/shared/Link/index.tsx

```diff
@@ -123,10 +123,15 @@ export const Link = forwardRef<HTMLAnchorElement, Props>(function Link(
 
   if (useNextLink && href != null) {
     return (
-      <NextLink href={href}>
-        {/* eslint-disable-next-line */}
-        <a {...restOfProps} ref={ref} target={target} onClick={handleClick} />
-      </NextLink>
+      <NextLink
+        {...restOfProps}
+        ref={ref}
+        target={target}
+        onClick={handleClick}
+        href={href}
+        // Next.js v12 has legacyBehavior set to true by default
+        legacyBehavior={false}
+      />
```

Comment:
We might need a shim for `next/image`, but we don't need it for `next/link`.

`legacyBehavior` is supported from [Next.js 12](https://github.com/vercel/next.js/pull/36436). So this should work for both Next.js 13 and 12.

Reply:
Thanks for linking the shim code from React! We probably need it for `next/image`

Reply:
It seems this was for Next.js v12.2: https://nextjs.org/blog/next-12-2#other-improvements

We might still need to shim it if the version is lower. Or do you think we only support Next.js v12.3 and v13+?

Reply:
Our runtime has peerDependency of `"next": ">=12.2.0"`, so I thought that we only support Next v12.2+?

Reply:
Ok, perfect. I think that we do 12.2+ for on-demand ISR. This means our peer dependencies should be `"next": ">=12.2.0 < 13.0.0 || ^13.0.0"

Reply:
Cool, I added the next.js version checking, and package transpile support on Next.js 13 using their built-in feature.

Reply:
If they are on version `13.0.*`, there would be a warning:
```
warn  - You have enabled experimental feature (transpilePackages) in next.config.js.
```
because `transpilePackages` is still experimental. This should be fine, as on their MR they called it ["Move transpilePackages out of experimental"](https://github.com/vercel/next.js/pull/44194), but we could just not support `13.0.*` if we want to.